octokit/connection.py

```diff
@@ -0,0 +1,78 @@
+from .resources import Resource
+
+class Connection(object):
+
+  # Make a HTTP GET request
+  #
+  # @return Resource
+  def get(self, url, *args, **kwargs):
+    return self.request('GET', url, *args, **kwargs)
+
+  # Make a HTTP POST request
+  #
+  # @return Resource
+  def post(self, url, *args, **kwargs):
+    return self.request('POST', url, *args, **kwargs)
+
+  # Make a HTTP PUT request
+  #
+  # @return Resource
+  def put(self, url, *args, **kwargs):
+    return self.request('PUT', url, *args, **kwargs)
+
+  # Make a HTTP PATCH request
+  #
+  # @return Resource
+  def patch(self, url, *args, **kwargs):
+    return self.request('PATCH', url, *args, **kwargs)
+
+  # Make a HTTP DELETE request
+  #
+  # @return Resource
+  def delete(self, url, *args, **kwargs):
+    return self.request('DELETE', url, *args, **kwargs)
+
+  # Make a HTTP HEAD request
+  #
+  # @return Resource
+  def head(self, url, *args, **kwargs):
+    return self.request('HEAD', url, *args, **kwargs)
+
+  # Continue following the relations until there are no more links
+  #
+  # *args           - Uri template argument
+  # **kwargs        – Uri template arguments
+  # Returns Resource
+  def paginate(self, url, *args, **kwargs):
+    params = {}
+    if 'per_page' in kwargs:
+      params['per_page'] = kwargs['per_page']
+      del kwargs['per_page']
+    elif self.auto_paginate:
+      # if per page is not defined, default to 100 per page
+      params['per_page'] = 100
+
+    if 'page' in kwargs:
+      params['page'] = kwargs['page']
+      del kwargs['page']
+
+    kwargs['params'] = params
+    resource = self.get(url, *args, **kwargs)
+    data = list(resource.schema)
+
+    if self.auto_paginate:
+      while 'next' in resource.rels and self.rate_limit.remaining > 0:
```

Comment:
So for convenience method mixins, like Users, they would update/get the rate limit by accessing it like this too?